ruby_event_store/lib/ruby_event_store/in_memory_repository.rb

```diff
@@ -80,7 +71,24 @@ def normalize_to_array(events)
       return *events
     end
 
-    def append_with_synchronize(events, expected_version, stream, stream_name)
+    def add_to_stream(events, expected_version, stream_name, include_global)
+      raise InvalidExpectedVersion if !expected_version.equal?(:any) && stream_name.eql?(GLOBAL_STREAM)
+      events = normalize_to_array(events)
+      stream = read_stream_events_forward(stream_name)
+      expected_version = case expected_version
```

Comment:
why this code (with expected version) is repeated in another place?

Reply:
Where?

Reply:
One is InMemoryRepository and the other is ActiveRecordRepository.

Reply:
sure but is exactly same logic, so why to not share it from ruby_event_store? Maybe extract to separate class all versioning code as strategy pattern, and use it in both places

Reply:
It differs in 1 line and it is only repeated in 2 places. Once the number of places reaches 3 or 4 I will consider sharing it between various implementations. So far I was not inclined to do it.